public/pages/CreateHistoricalDetector/utils/helpers.ts

```diff
@@ -0,0 +1,134 @@
+  values: HistoricalDetectorFormikValues,
+  detector: Detector
+): Detector {
+  let apiRequest = {
+    ...detector,
+    name: values.name,
+    description: values.description,
+    indices: values.index.map((index) => index.label),
+    timeField: values.timeField,
+    featureAttributes: formikToFeatures(values.featureList, false),
+    uiMetadata: {
+      ...detector?.uiMetadata,
+      features: { ...formikToUIMetadata(values.featureList) },
+    },
+    detectionInterval: {
+      period: { interval: values.detectionInterval, unit: UNITS.MINUTES },
+    },
+    windowDelay: {
+      period: { interval: values.windowDelay, unit: UNITS.MINUTES },
+    },
+    shingleSize: values.shingleSize,
+    detectionDateRange: {
+      startTime: convertTimestampToNumber(values.startTime),
+      endTime: convertTimestampToNumber(values.endTime),
+    },
+  } as Detector;
+
+  return apiRequest;
+}
+
+export function historicalDetectorToFormik(
+  detector: Detector
+): HistoricalDetectorFormikValues {
+  const initialValues = cloneDeep(INITIAL_HISTORICAL_DETECTOR_VALUES);
+  if (isEmpty(detector)) {
+    return initialValues;
+  }
+  return {
+    ...initialValues,
+    name: detector.name,
+    description: detector.description,
+    index: [{ label: detector.indices[0] }],
+    timeField: detector.timeField,
+    featureList: featuresToFormik(detector),
+    detectionInterval: get(detector, 'detectionInterval.period.interval', 10),
+    shingleSize: get(detector, 'shingleSize', SHINGLE_SIZE),
+    startTime: get(detector, 'detectionDateRange.startTime'),
+    endTime: get(detector, 'detectionDateRange.endTime'),
+  };
+}
+
+function featuresToFormik(detector: Detector): FeaturesFormikValues[] {
+  const featureUiMetaData = get(detector, 'uiMetadata.features', []);
+  const features = get(detector, 'featureAttributes', []);
+  // @ts-ignore
+  return features.map((feature: FeatureAttributes) => {
+    return {
+      ...featureUiMetaData[feature.featureName],
+      ...feature,
+      aggregationQuery: JSON.stringify(feature['aggregationQuery'], null, 4),
+      aggregationOf: get(
+        featureUiMetaData,
+        `${feature.featureName}.aggregationOf`
+      )
+        ? [
+            {
+              label: get(
+                featureUiMetaData,
+                `${feature.featureName}.aggregationOf`
+              ),
+            },
+          ]
+        : [],
+      featureType: get(featureUiMetaData, `${feature.featureName}.featureType`)
+        ? get(featureUiMetaData, `${feature.featureName}.featureType`)
+        : FEATURE_TYPE.CUSTOM,
+    };
+  });
+}
+
+export function convertTimestampToString(timestamp: number | string) {
+  if (typeof timestamp === 'string') {
+    return timestamp;
+  }
+  return moment(timestamp).format();
+}
+
+export function convertTimestampToNumber(timestamp: number | string) {
+  if (typeof timestamp === 'string') {
+    return datemath.parse(timestamp)?.valueOf();
+  }
+  return timestamp;
+}
```

Comment:
minor: move those 2 methods to a more common [place](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/master/public/utils/utils.tsx), since they are very helpful and can be reused by other components as well.

Reply:
Good idea, refactored to `public/utils/utils.tsx`